.circleci/config.yml

```diff
@@ -20,16 +20,170 @@ jobs:
           command: |
             cd ../
             mkdir -p ~/.ssh && touch ~/.ssh/config && echo $'Host github.com\n\tStrictHostKeyChecking no' > ~/.ssh/config
+      
+      # Download and cache dependencies
+
+      # libs
+      - restore_cache:
+          name: Restore libs cache
+          keys:
+          - v1-libs-{{ checksum "libs/package-lock.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-libs-
+      - run: rm -rf /home/circleci/project/libs/node_modules/websocket/.git
+      - run: 
+          name: libs repo init
+          command: |
+            cd libs
+            npm install
+      - save_cache:
+          name: Save libs cache
+          key: v1-libs-{{ checksum "libs/package-lock.json" }}
+          paths:
+            - libs/node_modules
+
+      # data contracts
+      - restore_cache:
+          name: Restore contracts cache 
+          keys:
+          - v1-data-contracts-{{ checksum "contracts/package-lock.json" }}
+          - v1-data-contracts- 
+      - run:
+          name: contract repo init
+          command: |
+            cd contracts/
+            npm install
+      - save_cache:
+          name: Save contracts cache
+          paths:
+            - contracts/node_modules
+          key: v1-data-contracts-{{ checksum "contracts/package-lock.json" }}
+
+      # eth contracts
+      - restore_cache:
+          name: Restore eth-contracts cache
+          keys:
+          - v1-eth-contracts-{{ checksum "eth-contracts/package-lock.json" }}
+          - v1-eth-contracts-
+      - run:
+          name: eth-contracts repo init
+          command: |
+            cd eth-contracts/
+            rm -rf node_modules/web3-providers-ws/node_modules/websocket/.git/
+            source ~/.bash_profile
+            nvm install 10.16
+            npm install
+      - save_cache:
+          name: Save eth-contracts cache
+          paths:
+            - eth-contracts/node_modules
+          key: v1-eth-contracts-{{ checksum "eth-contracts/package-lock.json" }}
+
+      # content node
+      - restore_cache:
+          name: Restore content-node cache
+          keys:
+          - v1-content-node-{{ checksum "creator-node/package-lock.json" }}
+          - v1-content-node- 
+      - run:
+          name: content-node repo init
+          command: |
+            source ~/.bash_profile
+            nvm install v10.16
+            cd creator-node/
+            npm install
+      - save_cache:
+          name: Save content-node cache
+          paths:
+            - creator-node/node_modules
+          key: v1-content-node-{{ checksum "creator-node/package-lock.json" }}
+
+      # discovery node -- slightly trickier. will revisit
```

Comment:
`init-repos.sh` actually does not even install discovery deps. can probably remove this? @hareeshnagaraj